### DISCUSSION TITLE
Dirty Fields and useFieldsArray

### DISCUSSION THREAD

**Discussion by Daneng66** [react-hook-form/react-hook-form#7575] [General] [answered]
Is it a known behaviour that changing the indexes of records in a field array (e.g. via append or delete) will report fields as dirty even though the data behind the record hasn't changed?

**Comment by Moshyfawn** [comment, marked as answer]
I believe so, yeah. By manipulating array fields you're updating the array field value _(adding / removing / moving elements inside the array)_. Only the newly appended / removed value is marked as dirty. See[ this CSB](https://codesandbox.io/s/dreamy-minsky-gg37x?file=/src/App.tsx)
#6448 discussion might be related, check it out

**Reply by Daneng66** [reply]
That's fine I won't create a bug then, it just means that I have to sanitise the dirty fields information and check equality before creating my patch requests to the server.

**Reply by Daneng66** [reply]
BTW really appreciate how quickly you guys reply  🙌

**Reply by Moshyfawn** [reply]
You can also try `reset` your form with the current values after manipulating the array _(`append` / `remove` etc)_; it will set all fields to unmodified (`isDirty: false`)

**Reply by Daneng66** [reply]
> You can also try `reset` your form with the current values after manipulating the array _(`append` / `remove` etc)_; it will set all fields to unmodified (`isDirty: false`)

I don't rely on isDirty, I only look at the content of dirtyFields to determine if I can submit and what data has changed

**Reply by Moshyfawn** [reply]
`reset` resets `dirtyFields` form state as well. See [this CSB](https://codesandbox.io/s/condescending-grass-qptgt?file=/src/App.tsx). 
1. Append a new field
2. See at the bottom of the form "Dirty Fields" is empty, "Values" has a new field value added
